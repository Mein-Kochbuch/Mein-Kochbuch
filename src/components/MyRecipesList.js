import React from "react";
import {View} from "react-native";
import MultiListItem from "./MultiListItem";
import CookbookListItem from "./CookbookListItem";

export default function MyRecipesList({cookbooks}) {

    return (
        <View>
            <MultiListItem title={"All Recipes"} icon="all"/>
            <MultiListItem title={"Favorites"} icon="heart"/>
            <View style={{margin: 10}}/>
            {
                cookbooks.map((cookbook) => <CookbookListItem key={cookbook.pk} cookbook={cookbook}/>)
            }
        </View>
    )
}
