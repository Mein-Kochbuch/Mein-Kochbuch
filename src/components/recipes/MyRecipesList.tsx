import React from "react";
import {View} from "react-native";
import MultiListItem from "../list/MultiListItem";
import CookbookListItem from "../cookbooks/CookbookListItem";
import styled from "styled-components/native";
import {Cookbook} from "../../models/Cookbook";

interface MyRecipeListProps {
    cookbooks: Cookbook[]
}

export default function MyRecipesList({cookbooks}: MyRecipeListProps) {

    return (
        <View>
            <MultiListItem title={"All Recipes"} icon="all"/>
            <MultiListItem title={"Favorites"} icon="heart"/>
            <Spacer/>
            {
                cookbooks.map((cookbook) => <CookbookListItem key={cookbook.pk} cookbook={cookbook}/>)
            }
        </View>
    )
}

const Spacer = styled.View`
  margin: 12px;
`
