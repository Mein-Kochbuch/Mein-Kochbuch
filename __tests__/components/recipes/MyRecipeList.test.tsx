import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MyRecipesList, {
  MyRecipeListProps,
} from '../../../src/components/recipes/MyRecipesList';

jest.mock('../../../src/components/list/MultiListItem', () => {
  return props => {
    return <div {...props} />;
  };
});

jest.mock('../../../src/components/cookbooks/CookbookListItem', () => {
  return props => {
    return <div {...props} />;
  };
});

describe('MyRecipeList', () => {
  it('MyRecipesList Test', () => {
    const myRecipesListProps: MyRecipeListProps = {
      cookbooks: [
        {
          id: '1',
          name: 'cookbook-1',
          owner: {
            id: '1',
            name: 'me',
          },
          privacy: false,
          recipes: [],
          thumbnail: {
            id: '1',
            url: 'cookbook-thumbnail-url',
          },
        },
        {
          id: '2',
          name: 'cookbook-2',
          owner: {
            id: '2',
            name: 'you',
          },
          privacy: false,
          recipes: [],
          thumbnail: {
            id: '2',
            url: 'cookbook-thumbnail-url-2',
          },
        },
      ],
    };
    const component = renderer.create(
      <MyRecipesList {...myRecipesListProps} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
