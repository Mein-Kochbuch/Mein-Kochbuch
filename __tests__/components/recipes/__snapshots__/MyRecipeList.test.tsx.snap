// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipeList MyRecipesList Test 1`] = `
<View>
  <div
    icon="all"
    title="All Recipes"
  />
  <div
    icon="heart"
    title="Favorites"
  />
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  />
  <div
    cookbook={
      Object {
        "id": "1",
        "name": "cookbook-1",
        "owner": Object {
          "id": "1",
          "name": "me",
        },
        "privacy": false,
        "recipes": Array [],
        "thumbnail": Object {
          "id": "1",
          "url": "cookbook-thumbnail-url",
        },
      }
    }
  />
  <div
    cookbook={
      Object {
        "id": "2",
        "name": "cookbook-2",
        "owner": Object {
          "id": "2",
          "name": "you",
        },
        "privacy": false,
        "recipes": Array [],
        "thumbnail": Object {
          "id": "2",
          "url": "cookbook-thumbnail-url-2",
        },
      }
    }
  />
</View>
`;

exports[`MyRecipesList Test 1`] = `
<View>
  <div
    icon="all"
    title="All Recipes"
  />
  <div
    icon="heart"
    title="Favorites"
  />
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  />
  <div
    cookbook={
      Object {
        "cookbook": "1",
        "id": "1",
      }
    }
  />
  <div
    cookbook={
      Object {
        "cookbook": "2",
        "id": "2",
      }
    }
  />
</View>
`;
